src/dali_executor/executor.test.cc

```diff
@@ -86,4 +86,56 @@ TEST_CASE("Scaling Pipeline")
   }
 }
 
+TEST_CASE("RN50 pipeline")
+{
+  std::string pipeline(
+      (const char*)pipelines::rn50_gpu_dali_chr,
+      pipelines::rn50_gpu_dali_len);
+  DaliExecutor executor(pipeline, 0);
+  IODescr<false> input;
+  input.name = "DALI_INPUT_0";
+  input.type = dali_data_type_t::DALI_UINT8;
+  input.shape = TensorListShape<1>(1);
+  input.shape.set_tensor_shape(0, TensorShape<>(data::jpeg_image_len));
+  input.buffer = span<char>((char*)(data::jpeg_image_str),
+                            data::jpeg_image_len);
+  input.device = device_type_t::CPU;
+
+  auto execute_with_image = [&]() {
+    auto output = executor.Run(std::vector<IODescr<false>>({input}));
+    std::vector<float> output_buffer(output[0].shape.num_elements());
+    std::vector<IODescr<false>> output_vec(1);
+    auto& outdesc = output_vec[0];
+    outdesc.device = device_type_t::GPU;
+    outdesc.device_id = 0;
+    outdesc.buffer = make_span(
+        (char*)output_buffer.data(),
+        output_buffer.size() * sizeof(decltype(output_buffer)::size_type));
+    executor.PutOutputs(output_vec);
```

Comment:
Can we validate the output anyhow?

Reply:
Hard to say. I could validate the shape but I'm not sure about contents as it's Resnet pipeline run on some random small image. The purpose of this test was rather to check for exceptions

Reply:
If the image is the same, I guess we can expect rather quite accurate Top1 or at least Top5.

Reply:
The image was single (0, 0, 0) pixel, so now I verify the output values.